### ISSUE TITLE
Example iOS is not work : (

### ISSUE THREAD

**Issue by AngYuDet** [shogo4405/SRTHaishinKit.swift#60]
**Describe the bug**
Example iOS is not work : (
  
**To Reproduce**
1. git clone https://github.com/shogo4405/SRTHaishinKit.swift
2. cd SRTHaishinKit.swift
3. carthage update --use-xcframeworks --platform iOS
4. open xcode, change the "url" string to my srt server in Preference.swift file
5. build it and on my iphone 12 pro max, i click the "publish button" which at the ower right corner of the screen
6. but when ffplay it，no stream is on

**Expected behavior**
The app can publish a srt stream

**Screenshots**
1. connect success log in xcode console:
<img width="404" alt="image" src="https://github.com/shogo4405/SRTHaishinKit.swift/assets/135143957/1493179a-ba69-4de6-9dcd-c75b83018d8d">

2. ffplay it, but no stream frame to play
<img width="488" alt="image" src="https://github.com/shogo4405/SRTHaishinKit.swift/assets/135143957/9d8b6934-459d-4d56-8c4b-55ed01c54cd6">

**Desktop (please complete the following information):**
 - OS: iOS 16.1.1
 
**Smartphone (please complete the following information):**
 - Device: iphone 12 pro max

**Additional context**
None

**Comment by AngYuDet** [comment]
I'm a new ios developer, but i guess maybe the example dont handle the srt streamid with secret key.

Like the url:

    srt://xx.xx.xx.xx:10080?streamid=#!::r=live/abekcd?secret=xxx,m=publish

can sucess published in obs.

**Comment by shogo4405** [comment]
The provided URL doesn't seem to be displayed correctly... There might be a possibility that the parsing is not working properly. I'm using URL(string:) as part of the internal processing. It tends to ignore URLs that are not considered valid.

**Comment by AngYuDet** [comment]
Yes, you are right, see the url srt://xx.xx.xx.xx:10080?streamid=#!::r=live/abekcd?secret=xxx,m=publish : 
    
    1. '#!::' stand for yaml format
    2. 'r=' stand for rtmp addr map
    3. 'secret=' to auth
    4. 'm=' means just publishing stream

And this is maybe the srs(Simple Realtime Server) server private protocol(i dont know srt protocol well) which obs have to adapt to.

So the simple URL PARSING in the examples dont work.

**Comment by shogo4405** [comment]
Since it seems unlikely that I can resolve it, I will close it. The PR is still welcome.
